src/components/swap/components/select-token-modal.tsx

```diff
@@ -28,11 +29,13 @@ type SelectTokenModalProps = {
 export const SelectTokenModal = (props: SelectTokenModalProps) => {
   const { isOpen, onClose, onSelectedToken, tokens } = props
   const isDarkMode = props.isDarkMode ?? false
+  const tokenAddressField = props.tokenAddressField ?? 'address'
   const showBalances = props.showBalances ?? true
   const tokenAddresses = useMemo(
-    () => tokens.map((token) => token.address!),
-    [tokens],
+    () => tokens.map((token) => token[tokenAddressField] ?? ''),
+    [tokens, tokenAddressField],
```

Comment:
great find! another place where taking chainId into account was missing. so we can just leverage existing functions:

<img width="554" alt="Bildschirmfoto 2024-05-30 um 09 18 05" src="https://github.com/IndexCoop/index-app/assets/2104965/5aa0ccf3-d6db-4bff-8c78-af9dea8951a7">